agent/structs/config_entry_routes.go

```diff
@@ -432,6 +435,18 @@ type URLRewrite struct {
 	Path string
 }
 
+type RetryFilter struct {
+	NumRetries            *uint32
+	RetryOn               []string
+	RetryOnStatusCodes    []uint32
+	RetryOnConnectFailure *bool
```

Comment:
I suspect we can actually remove `RetryOnConnectFailure` based on [this code + comment](https://github.com/hashicorp/consul/blob/284e3bdb547915d82bfc89296ccad485befdee59/agent/xds/routes.go#L764C1-L776C3) pointing out that there are two ways of getting `"connect-failure"` added to the `RetryOn` list. The standalone field is referred to as "legacy".

I'm fine leaving for now since our code is already built to handle both, but I'd like to discuss with the team prior to release whether we want to drop the "legacy" option.